src/main/java/uk/gov/hmcts/reform/userprofileapi/service/impl/DeleteUserProfileServiceImpl.java

```diff
@@ -67,15 +120,31 @@ public UserProfilesDeletionResponse deleteUserProfiles(List<UserProfile> userPro
 
     /**
      * This method is used to find the user profile exist or not with the give userId.
-     *
      */
-    public  UserProfile validateUserStatus(String userId) {
+    public UserProfile validateUserStatus(String userId) {
         Optional<UserProfile> userProfileOptional = userProfileRepository.findByIdamId(userId.trim());
-        if (!userProfileOptional.isPresent()) {
-            throw new ResourceNotFoundException("could not find user profile for userId:" + userId);
+        if (userProfileOptional.isEmpty()) {
+            throw new ResourceNotFoundException("could not find user profile for userId: " + userId);
         }
         return userProfileOptional.get();
     }
 
-}
+    public UserProfilesDeletionResponse validateUserAfterIdamDelete(
+            Optional<UserProfile> userProfile, String userId, int status) {
+        UserProfilesDeletionResponse deletionResponse = new UserProfilesDeletionResponse();
+
+        if (userProfile.isPresent()) {
+            return deleteUserProfiles(singletonList(userProfile.get()));
+
+        } else if (status == 204) {
```

Comment:
Later, we can change it to HttpStatus.NOT_FOUND.value()

Reply:
Addressed here: https://github.com/hmcts/rd-user-profile-api/pull/664/commits/7f2c50137b6483af001a68deb3daec703d5d3e55